### ISSUE TITLE
4.5.8 Compilation error with PHP 8

### ISSUE THREAD

**Issue by luispabon** [swoole/swoole-src#3871]
Please answer these questions before submitting your issue. Thanks!

1. What did you do? If possible, provide a simple script for reproducing the error.
https://github.com/phpdocker-io/base-images/blob/master/php/8.0/swoole/Dockerfile

2. What did you expect to see?
No compilation issues

3. What did you see instead?
```
/bin/bash /tmp/swoole-src/libtool --mode=compile g++ -I. -I/tmp/swoole-src -I/tmp/swoole-src/include -I/tmp/swoole-src/main -I/tmp/swoole-src -I/usr/include/php/20200930 -I/usr/include/php/20200930/main -I/usr/include/php/20200930/TSRM -I/usr/include/php/20200930/Zend -I/usr/include/php/20200930/ext -I/usr/include/php/20200930/ext/date/lib -I/tmp/swoole-src -I/tmp/swoole-src/include -I/tmp/swoole-src/ext-src -I/tmp/swoole-src/thirdparty/hiredis  -DHAVE_CONFIG_H  -g -O2 -Wall -Wno-unused-function -Wno-deprecated -Wno-deprecated-declarations -std=c++11    -c /tmp/swoole-src/ext-src/php_swoole.cc -o ext-src/php_swoole.lo 
libtool: compile:  g++ -I. -I/tmp/swoole-src -I/tmp/swoole-src/include -I/tmp/swoole-src/main -I/tmp/swoole-src -I/usr/include/php/20200930 -I/usr/include/php/20200930/main -I/usr/include/php/20200930/TSRM -I/usr/include/php/20200930/Zend -I/usr/include/php/20200930/ext -I/usr/include/php/20200930/ext/date/lib -I/tmp/swoole-src -I/tmp/swoole-src/include -I/tmp/swoole-src/ext-src -I/tmp/swoole-src/thirdparty/hiredis -DHAVE_CONFIG_H -g -O2 -Wall -Wno-unused-function -Wno-deprecated -Wno-deprecated-declarations -std=c++11 -c /tmp/swoole-src/ext-src/php_swoole.cc  -fPIC -DPIC -o ext-src/.libs/php_swoole.o

In file included from /tmp/swoole-src/ext-src/php_swoole_cxx.h:19,
                 from /tmp/swoole-src/ext-src/php_swoole.cc:16:

/tmp/swoole-src/ext-src/php_swoole.h: In function 'zend_bool sw_zend_is_callable_at_frame(zval*, zval*, zend_execute_data*, uint, char**, size_t*, zend_fcall_info_cache*, char**)':
/tmp/swoole-src/ext-src/php_swoole.h:918:11: error: 'zend_is_callable_at_frame' was not declared in this scope; did you mean 'sw_zend_is_callable_at_frame'?
  918 |     ret = zend_is_callable_at_frame(zcallable, zobject ? Z_OBJ_P(zobject) : NULL, frame, check_flags, &name, fci_cache, error);
      |           ^~~~~~~~~~~~~~~~~~~~~~~~~
      |           sw_zend_is_callable_at_frame

make: *** [Makefile:209: ext-src/php_swoole.lo] Error 1
```

Full build: https://ci.auronconsulting.co.uk/teams/main/pipelines/phpdocker-base-images/jobs/php-8.0/builds/56

4.5.7 compiled correctly, as you can see on the build history on the link above.

4. What version of Swoole are you using (show your `php --ri swoole`)?
4.5.8

5. What is your machine environment used (show your `uname -a` & `php -v` & `gcc -v`) ?

php 8.0.0RC3
uname -a: n/a (docker)
gcc version 9.3.0 (Ubuntu 9.3.0-17ubuntu1~20.04)

**Comment by haohetao** [comment]
using master
Php8 is not supported in all release versions

**Comment by doubaokun** [comment]
@luispabon This will be fixed when PHP releases the official 8.0 or on the master branch of PHP.
